### ISSUE TITLE
why the position is bad 

### ISSUE THREAD

**Issue by thierryH91200** [emankovski/CollapseSplitViews#1]
I took your example if you could help me

[CollapseSplitViews-master.zip](https://github.com/emankovski/CollapseSplitViews/files/1584104/CollapseSplitViews-master.zip)

thanks

**Comment by emankovski** [comment]
What is going on?

**Comment by thierryH91200** [comment]
@IBAction func toggleButtonPressed(_ sender: Any) {

        print( splitView.bounds.width)
        if splitView.isSubviewCollapsed(splitView.arrangedSubviews[2]) {
            splitView.setPosition(1 * splitView.bounds.width / 3, ofDividerAt: 0)
            splitView.setPosition(2 * splitView.bounds.width / 3, ofDividerAt: 1)
        } else {
            print( 1 * splitView.bounds.width / 3 )

            splitView.setPosition(1 * splitView.bounds.width / 3, ofDividerAt: 0)
            splitView.setPosition(splitView.bounds.width, ofDividerAt: 1)
        }
    }

the position of index is 1/3 of splitView.bounds.width at index 0 in my example 
**and that's exactly half**

look at the zip


![capture d ecran 2017-12-23 a 14 59 26](https://user-images.githubusercontent.com/6251463/34320111-f109d5f8-e7f1-11e7-9225-9b6381fcd7de.jpg)

**Comment by emankovski** [comment]
Yes but do you consider gutter width as well? You probably need to subtract gutter width.

**Comment by thierryH91200** [comment]
I don't think   

 @IBAction func toggleButtonPressed(_ sender: Any) {

        print( splitView.bounds.width)
        if splitView.isSubviewCollapsed(splitView.arrangedSubviews[2]) {
            splitView.setPosition(1 * splitView.bounds.width / 3, ofDividerAt: 0)
            splitView.setPosition(2 * splitView.bounds.width / 3, ofDividerAt: 1)
        } else {
            print( 2 * splitView.bounds.width / 3 )

            **splitView.setPosition(2 * splitView.bounds.width / 3, ofDividerAt: 0)**
            splitView.setPosition(splitView.bounds.width, ofDividerAt: 1)
        }
    }

in this example 
the position of index is 2/3 of splitView.bounds.width


![capture d ecran 2017-12-23 a 15 23 45](https://user-images.githubusercontent.com/6251463/34320237-5086ef86-e7f5-11e7-9b97-2dd66ea2f994.jpg)

**Comment by emankovski** [comment]
ok I see. The problem is that it cannot squeeze something because of missing constraints:


_2017-12-23 09:29:10.179592 CollapseSplitViews[39139:23336082] [Layout] Detected missing constraints for <NSSplitDividerView: 0x600000162a00>.  It cannot be placed because there are not enough constraints to fully define the size and origin. Add the missing constraints, or set translatesAutoresizingMaskIntoConstraints=YES and constraints will be generated for you. If this view is laid out manually on macOS 10.12 and later, you may choose to not call [super layout] from your override. Set a breakpoint on DETECTED_MISSING_CONSTRAINTS to debug. This error will only be logged once._

As soon as you see such message in console it means layout is broken.

**Comment by emankovski** [comment]
There seems to be a problem with gutter 0. If I comment the line, the position is ok:
![image](https://user-images.githubusercontent.com/15099166/34320358-6cf49378-e7c6-11e7-87e2-922f681f8634.png)

Yea not sure what is going on here. Sorry. I have not been doing mac development for almost a year now.

**Comment by thierryH91200** [comment]
thanks anyway but I do not understand
we can not put the gutter or we want
I do not have a problem of constraint
